### PR TITLE
OCP-DEMO Upgrades Wildfly 13 - OGM 5.4.0.Beta2 - Infinispan 9.3.0.Final

### PR DESCRIPTION
**Pull request by fax4ever**

### CODE REVIEW

File: openshift/message-board/account-micro/src/main/java/org/hibernate/demo/account/User.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+@Entity
+public class User {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE)
+	private Long id;
+
+	@NotEmpty
+	@Column(unique = true)
+	private String userName;
```
Comment (gsmet) [inline openshift/message-board/account-micro/src/main/java/org/hibernate/demo/account/User.java]: I would add a `@NaturalId` here.

File: openshift/message-board/account-micro/src/main/java/org/hibernate/demo/account/UserRepo.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	public void add(@Valid User user) {
+		em.persist( user );
+	}
+
+	public User findByUserName(String userName) {
+		Query query = em.createQuery( "from User u where u.userName = :userName" );
+		query.setParameter( "userName", userName );
+		return (User) query.getSingleResult();
```
Comment (gsmet) [inline openshift/message-board/account-micro/src/main/java/org/hibernate/demo/account/UserRepo.java]: I would use the natural id created previously. This allows to use the 2nd level cache if set.

File: openshift/message-board/account-micro/src/main/java/org/hibernate/demo/account/UserRepo.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	public void delete(String userName) {
+		Query query = em.createQuery( "delete from User u where u.userName = :userName" );
+		query.setParameter( "userName", userName );
+		query.executeUpdate();
```
Comment (gsmet) [inline openshift/message-board/account-micro/src/main/java/org/hibernate/demo/account/UserRepo.java]: I don't think this is 2nd level cache friendly. I would do a load then a delete.

That makes sense as I think, in a real application, you should probably return an error if you're trying to delete a user who does not exist.

File: openshift/message-board/message-service/pom.xml [hunk 4]
```diff
@@ -135,6 +132,13 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <!-- TODO Workaround to fix wildfly-arquillian-container-managed \https://issues.jboss.org/browse/WFLY-9747 -->
```
Comment (gsmet) [inline openshift/message-board/message-service/pom.xml]: There is a `\` at the beginning of the URL, it should be removed I think.

File: openshift/message-board/message-service/src/main/java/org/hibernate/demo/message/post/core/service/MessageService.java [hunk 5]
```diff
@@ -58,7 +58,7 @@
 
 	@POST
 	@Consumes( MediaType.APPLICATION_JSON )
-	public void addMessage( @Valid Message message ) {
+	public void addMessage( Message message ) {
```
Comment (gsmet) [inline openshift/message-board/message-service/src/main/java/org/hibernate/demo/message/post/core/service/MessageService.java]: Why did you remove that?

File: openshift/message-board/account-micro/pom.xml [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    <description>Preview of Account MicroService - using Thorntail v4</description>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+
+        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>1.8</maven.compiler.source>
+
+        <!-- TODO: looking forward to 4.0.0.Final -->
+        <thorntail.version>4.0.0-SNAPSHOT</thorntail.version>
```
Comment (gsmet) [inline openshift/message-board/account-micro/pom.xml]: You used the `version.<project>` convention elsewhere but not here. We should try to be consistent across the whole project.

File: openshift/message-board/README.asciidoc [hunk 7]
```diff
@@ -65,4 +70,19 @@ http://web-message-board.[YOUR-IP].nip.io/account-service/user
...
+It might need to downgrade angular-cli version:
+----
+sudo -s npm uninstall -g @angular/cli
+sudo -s npm install -g @angular/cli@6.0.8
+----
+For more information see https://github.com/angular/angular-cli/issues/11661[Issue Page].
+
+### MiniShift CDK v3.5.0-1
+Actually there is an issue with MiniShift v1.20.0+1bd6d5c, CDK v3.5.0-1.
+The persistent volume for Infinispan server pod is mounted as root user.
```
Comment (gsmet) [inline openshift/message-board/README.asciidoc]: Is this the current version? Is it a real issue for the demo?

How do I fix it?

You're not talking about MiniShift before so it comes a bit out of the blue.

Comment (fax4ever) [reply openshift/message-board/README.asciidoc]: Yes it is in the current (last at the moment) version. Actually I don't know how to fix it, thus I'm using the Openshift Oring provided by Fedora 28, where the issue does not occur.

**Comment by fax4ever** [comment]
Many thanks @gsmet for the review.

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-demos/pull/32. thanks

**Comment by fax4ever** [comment]
Thank you @DavideD
